src_new/controllers/UAS/phoenix.js

```diff
@@ -0,0 +1,529 @@
+    //		empty array should be passed []
+    //		but then what should be passed ?
+    case 'adSizes':
+      if( dataAttributeFunctionMap.hasOwnProperty( dataAttribute ) ){
+        // temp will contain an array of sizes
+        temp = theSlot[ dataAttributeFunctionMap[ dataAttribute ] ]();
+        len = temp.length;
+        for(i=0; i<len; i++){
+          if(temp[i][0] && temp[i][1]){
+            output += temp[i][0] + 'x' + temp[i][1] + ','
+          }
+        }
+        output = output.substr(0, output.length -1 );
+      }
+      break;
+
+    case 'keywords':
+      if( dataAttributeFunctionMap.hasOwnProperty( dataAttribute ) ){
+        output = theSlot[ dataAttributeFunctionMap[ dataAttribute ] ]();
+        output = output.join();
+      }
+      break;
+
+    case 'keywordsOperation':
+      if( dataAttributeFunctionMap.hasOwnProperty( dataAttribute ) ){
+        output = theSlot[ dataAttributeFunctionMap[ dataAttribute ] ]();
+      }
+      break;
+
+      case 'targetings':
+      case 'extraParameters':
+        if( dataAttributeFunctionMap.hasOwnProperty( dataAttribute ) ){
+
+          // temp will contain list of keys first
+          temp = theSlot[ dataAttributeFunctionMap[ dataAttribute ] ]();
+          len = temp.length;
+          for(i=0; i<len; i++){
+
+            switch(dataAttribute) {
+              case 'targetings':
+                output += temp[i] + '=' + (theSlot[ dataAttributeFunctionMap[ 'targetingByKey' ] ]( temp[i] )).join() + '&';
+                break;
+
+              case 'extraParameters':
+                output += temp[i] + '=' + (theSlot[ dataAttributeFunctionMap[ 'extraPatameterByKey' ] ]( temp[i] )).join() + '&';
+                break;
+            }
+
+          }
+          output = output.substr(0, output.length -1 );
+        }
+        break;
+  }
+
+  return output;
+}
+
+function combineSlotsData(dataAttribute, arrayOfSlots) {
+  var theSlot,
+    output = "",
+    index,
+    aSlot,
+    defaultDataAttributeCombiner = "|",
+    //mention only exceptions for different dataAttributeCombiner
+    dataAttributeCombiner = {};
+
+  for(index in arrayOfSlots){
+    aSlot = arrayOfSlots[ index ];
+    output += combineSlotSpecificData(dataAttribute, aSlot) + (dataAttributeCombiner.hasOwnProperty(dataAttribute) ? dataAttributeCombiner[dataAttribute]: defaultDataAttributeCombiner);
+    aSlot.setStatus( CONSTANTS.SLOT_STATUS.DISPLAYED );
+  }
+  output = output.substr(0, output.length -1 );
+
+  return output;
+}
+
+function generateAdServerCall(arrayOfSlots, req_type, customInfo, queryParams){
+  var queryString = [],
+    currTime = new Date(),
+    key;
+
+  queryString.push(
+    'req_type=' + req_type,
+    'sec=' + window.PWT.secure,
+    'res_format=2',
+    //Callback function
+    'cback=window.parent.Phoenix.callback', // todo: window.parent ??  we may need to rethink
+    //Random Number
+    'rndn='			+ Math.random(),
+    'purl='			+ encodeURIComponent(window.PWT.pageURL),
```

Comment:
use Util function

Reply:
We use util.getMetaInfo(win) once in owt.js which calculates & returns big object & we assign its output to window.PWT.XX . So I think current implementation is right